src/bots/liquidator.ts

```diff
@@ -226,13 +234,47 @@ async function liqPerpPnl(
 		}
 	} else {
 		const start = Date.now();
-		driftClient
-			.liquidatePerpPnlForDeposit(
-				user.userAccountPublicKey,
-				user.getUserAccount(),
-				liquidateePosition.marketIndex,
-				depositMarketIndextoLiq,
-				depositAmountToLiq
+		const reserve = this.market.reserves.find(
+			(res) => res.config.liquidityToken.symbol === 'USDC'
+		);
+		await driftClient.txSender
+			.send(
+				new Transaction()
+					// this probably works
+					// and now; magik
+					.add(
+						flashBorrowReserveLiquidityInstruction(
+							depositAmountToLiq,
+							new PublicKey(reserve.config.liquidityAddress),
+							this.driftClient.getUser().getAssociatedTokenAddress(),
+							new PublicKey(reserve.config.address),
+							new PublicKey(this.market.config.address),
+							SOLEND_PRODUCTION_PROGRAM_ID
+						)
+					)
+					.add(
+						await driftClient.getLiquidatePerpPnlForDepositIx(
+							user.userAccountPublicKey,
+							user.getUserAccount(),
+							liquidateePosition.marketIndex,
+							depositMarketIndextoLiq,
+							depositAmountToLiq
+						)
+					)
```

Comment:
After this ix your account will have inherited the liquidated perp position. You will have to sell off the position via a call to `placeAndTake` , otherwise the account will need to hold enough collateral to repay the flashloan+fees